demo/src/app/components/typeahead/overview/sections/basic-usage/snippets.ts

```diff
@@ -0,0 +1,55 @@
+import {Snippet} from '../../common';
+
+
+
+export const SNIPPETS = {
+  observableSetup: {
+    template: Snippet({
+      language: 'html',
+      highlightedLines: '3',
+      code: `
+        <input
+          type="text"
+          [ngbTypeahead]="initializeTypeahead"
+        />
+      `,
+    }),
+    component: Snippet({
+      language: 'typescript',
+      highlightedLines: '2-3',
+      code: `
+        export class MyComponent {
+          initializeTypeahead = (input$: Observable<string>): Observable<string[]> =>
```

Comment:
I use the name `input$` because it comes from the `input` event of the Typeahead — to differentiate from `click` or `focus`. That can also be understood as "the content of the __input__ field". So I'd rather keep this name.